### ISSUE TITLE
[feature] Add MariaDB/MySQL support

### ISSUE THREAD

**Issue by EchedeyLR** [superseriousbusiness/gotosocial#728]
### Is your feature request related to a problem ?

Some deployments I have made are in servers using MySQL or MariaDB as database service.

It is uncommon to see more than one RDBMS being used at the same time and most applications already provide support for it, having not a real reason to move.

It would be great being able to take advantage of it instead of having to waste more resources or relying in SQLite which could not perform very well in mechanical storage as well as not being able to be integrated with DB backups in the same way that the others.

### Describe the solution you'd like.

Add option to setup a MariaDB/MySQL instance as database service for GoToSocial.

### Describe alternatives you've considered.

Maybe using an ORM could help in development? (I didn't check to see if you already use one)

### Additional context.

None by now.

**Comment by tsmethurst** [comment]
Heya! Sorry for the slow reply. I talked about this with the other developers and we reasoned that, because each db has its own peculiarities and implementation specifics, this is not something we have the bandwidth to support/maintain. It's already a bit of a pain managing the differences between sqlite and pg when doing database migrations and stuff, so adding even more database types is not really feasible currently. Sorry!
